packages/@aws-cdk/cfnspec/lib/canned-metrics.ts

```diff
@@ -0,0 +1,121 @@
+import { loadCannedMetricsFile, MetricTemplate } from './canned-metrics/canned-metrics-schema';
+
+export type NonEmptyArray<T> = [T, ...T[]];
+
+/**
+ * A single canned service metric
+ *
+ * These are kindly provided to us by the good people of CloudWatch Explorer.
+ */
+export interface CannedServiceMetric {
+  /**
+   * Metric namespace
+   */
+  readonly namespace: string;
+
+  /**
+   * Metric name
+   */
+  readonly metricName: string;
+
+  /**
+   * List of all possible dimension permutations for this metric
+   *
+   * Most metrics will have a single list of strings as their one set of
+   * allowed dimensions, but some metrics are emitted under multiple
+   * combinations of dimensions.
+   */
+  readonly dimensions: NonEmptyArray<string[]>;
```

Comment:
nit: I'd expect this to look like this:

```suggestion
  readonly dimensions: NonEmptyArray<string>;
```

Reply:
Ah but it definitely means what it says. It's an array of arrays.